src/HealthChecks.Azure.IoTHub/DependencyInjection/IoTHubHealthChecksBuilderExtensions.cs

```diff
@@ -23,22 +29,56 @@ public static class IoTHubHealthChecksBuilderExtensions
     /// <param name="tags">A list of tags that can be used to filter sets of health checks. Optional.</param>
     /// <param name="timeout">An optional <see cref="TimeSpan"/> representing the timeout of the check.</param>
     /// <returns>The specified <paramref name="builder"/>.</returns>
-    public static IHealthChecksBuilder AddAzureIoTHub(
+    public static IHealthChecksBuilder AddAzureIoTHubRegistryManager(
         this IHealthChecksBuilder builder,
-        Action<IoTHubOptions>? optionsFactory,
+        Func<IServiceProvider, RegistryManager>? registryManagerFactory = default,
+        Func<IServiceProvider, IotHubRegistryManagerOptions>? optionsFactory = default,
         string? name = default,
         HealthStatus? failureStatus = default,
         IEnumerable<string>? tags = default,
         TimeSpan? timeout = default)
     {
-        var options = new IoTHubOptions();
-        optionsFactory?.Invoke(options);
-
-        var registrationName = name ?? NAME;
+        return builder.Add(new HealthCheckRegistration(
+           name ?? NAME_REGISTRY_MANAGER,
+           sp => new IoTHubRegistryManagerHealthCheck(
+                registryManager: registryManagerFactory?.Invoke(sp) ?? sp.GetRequiredService<RegistryManager>(),
+                options: optionsFactory?.Invoke(sp)),
+           failureStatus,
+           tags,
+           timeout));
+    }
 
+    /// <summary>
+    /// Add a health check for Azure IoT Hub service client.
+    /// </summary>
+    /// <param name="builder">The <see cref="IHealthChecksBuilder"/>.</param>
+    /// <param name="serviceClientFactory">
+    /// /// An optional factory to obtain <see cref="ServiceClient" /> instance.
+    /// When not provided, <see cref="ServiceClient" /> is simply resolved from <see cref="IServiceProvider"/>.
+    /// </param>
+    /// <param name="optionsFactory">A action to configure the Azure IoT Hub connection to use.</param>
+    /// <param name="name">The health check name. Optional. If <c>null</c> the type name 'iothub' will be used for the name.</param>
+    /// <param name="failureStatus">
+    /// The <see cref="HealthStatus"/> that should be reported when the health check fails. Optional. If <c>null</c> then
+    /// the default status of <see cref="HealthStatus.Unhealthy"/> will be reported.
+    /// </param>
+    /// <param name="tags">A list of tags that can be used to filter sets of health checks. Optional.</param>
+    /// <param name="timeout">An optional <see cref="TimeSpan"/> representing the timeout of the check.</param>
+    /// <returns>The specified <paramref name="builder"/>.</returns>
+    public static IHealthChecksBuilder AddAzureIoTHubServiceClient(
+        this IHealthChecksBuilder builder,
+        Func<IServiceProvider, ServiceClient>? serviceClientFactory = default,
+        Func<IServiceProvider, IotHubServiceClientOptions>? optionsFactory = default,
+        string? name = default,
```

Comment:
nit: we can provide the default name since we know it

```suggestion
        string? name = NAME_SERVICE_CLIENT,
```